core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala

```diff
@@ -0,0 +1,169 @@
+      case wit: TransactionWitness =>
+        WitnessTransaction(version, inputs, outputsWithChange, lockTime, wit)
+    }
+
+    val dummyTxF = Transaction.addDummySigs(txNoChangeFee, inputInfos)
+
+    val txF = dummyTxF.map { dummyTx =>
+      val fee = feeRate.calc(dummyTx)
+      val change = totalCrediting - totalSpending - fee
+      val newChangeOutput = TransactionOutput(change, changeSPK)
+      val newOutputs = if (change <= Policy.dustThreshold) {
+        outputs
+      } else {
+        outputs :+ newChangeOutput
+      }
+
+      txNoChangeFee match {
+        case btx: NonWitnessTransaction =>
+          BaseTransaction(btx.version, btx.inputs, newOutputs, btx.lockTime)
+        case WitnessTransaction(version, inputs, _, lockTime, witness) =>
+          WitnessTransaction(version, inputs, newOutputs, lockTime, witness)
+      }
+    }
+
+    txF.flatMap { tx =>
+      val passInOutChecksT =
+        NonInteractiveWithChangeFinalizer.sanityDestinationChecks(
+          expectedOutPoints = inputs.map(_.previousOutput),
+          expectedOutputs = outputs,
+          changeSPK = changeSPK,
+          finalizedTx = tx)
+
+      val passChecksT = passInOutChecksT.flatMap { _ =>
+        Transaction.sanityChecks(forSigned = false,
+                                 inputInfos = inputInfos,
+                                 expectedFeeRate = feeRate,
+                                 tx = tx)
+      }
+
+      Future.fromTry(passChecksT.map(_ => tx))
+    }
+  }
+}
+
+object NonInteractiveWithChangeFinalizer {
+
+  def sanityDestinationChecks(
+      expectedOutPoints: Vector[TransactionOutPoint],
+      expectedOutputs: Vector[TransactionOutput],
+      changeSPK: ScriptPubKey,
+      finalizedTx: Transaction): Try[Unit] = {
+    //make sure we send coins to the appropriate destinations
+    val isMissingDestination =
+      !expectedOutputs.forall(finalizedTx.outputs.contains)
+    val hasExtraOutputs =
+      if (finalizedTx.outputs.size == expectedOutputs.size) {
+        false
+      } else {
+        //the extra output should be the changeOutput
+        !(finalizedTx.outputs.size == (expectedOutputs.size + 1) &&
+          finalizedTx.outputs.map(_.scriptPubKey).contains(changeSPK))
+      }
+    val spendingTxOutPoints = finalizedTx.inputs.map(_.previousOutput)
+    val hasExtraOutPoints =
+      !spendingTxOutPoints.forall(expectedOutPoints.contains)
+    if (isMissingDestination) {
+      TxBuilderError.MissingDestinationOutput
+    } else if (hasExtraOutputs) {
+      TxBuilderError.ExtraOutputsAdded
+    } else if (hasExtraOutPoints) {
+      TxBuilderError.ExtraOutPoints
+    } else {
+      Success(())
+    }
+  }
+
+  def txBuilderFrom(
+      outputs: Seq[TransactionOutput],
+      utxos: Seq[InputSigningInfo[InputInfo]],
+      feeRate: FeeUnit,
+      changeSPK: ScriptPubKey): RawTxBuilder = {
+    val inputs =
+      InputSigningInfo.calcSequenceForInputs(utxos, Policy.isRBFEnabled)
```

Comment:
I don't think RBF should be enabled by default. It is not good for privacy can cause double spend issues for txs before they are confirmed

Reply:
I'm open to the argument but my goal in this PR is to replace existing functionality, so this likely belongs on a different PR